### ISSUE TITLE
More granular location data as available from CT.gov?

### ISSUE THREAD

**Issue by p2** [opentrials/opentrials#517]
So far it seems location data is limited to the country (`locations.type: country`). Most of the time, trials have exact recruitment center data available on ClinicalTrials.gov (random example: https://clinicaltrials.gov/ct2/show/study/NCT02895360#contacts).

Am I missing something or is this data not in OpenTrials? Not yet?

(hope this is the right way to ask this kind of questions)

**Comment by vitorbaptista** [comment]
@p2 You're right, we don't have this information yet. We do collect it via our ClinicalTrials.gov scraper, but haven't been able to process it into our main database.

If you have some time and would like to give it a go, it should be relatively straightforward (at least to get the cities as well, the extra information like address and contacts will require some DB schema changes). We just need to change the function at https://github.com/opentrials/processors/blob/master/processors/nct/extractors.py#L114-L123.

**Comment by p2** [comment]
Alright great! Currently a bit time constrained, but I'll keep this on my list in case nobody else gets to it any time soon.

**Comment by p2** [comment]
Just to know, what is the sentiment towards DB schema changes? If this is an area to change the schema, it would be worthwhile to also think about geolocation, especially since elastic [has great support for it](https://www.elastic.co/guide/en/elasticsearch/guide/current/lat-lon-formats.html).

**Comment by vitorbaptista** [comment]
@p2 No problems to change the schema as needed. It's just better that, if you get the time to work on it, analyze and do write your suggestions before actually implementing it, so we make sure it's consistent with the rest of the codebase and can sync on them.

**Comment by p2** [comment]
That makes sense, will do!

**Comment by p2** [comment]
Finally got to take a closer look. Right now, the way OpenTrials supports `location` is quite limited and will be difficult to adapt to more granular information on the facility level, supporting things like recruitment status. It's only extracting country names found in `location_countries`, information that's also available in `location.facility.address.country`. Need to check whether one or the other is more complete.

IMHO the best approach would be to simlpy mirror what CT.gov is reporting for `locations`, which has a `facility` object (which includes and `address` that has `country`), a `status` object for recruitment status, and a `contact` element with site PI contact details.

**Comment by vitorbaptista** [comment]
@p2 Thanks for your time here. What changes to OpenTrials schema would you suggest to handle this? We also need to keep handling trials where the only data we have is the country, as some sources only provide this information.

**Comment by p2** [comment]
I'd suggest staying close to what ClinicalTrials.gov provides, the data they provide for each location does seem sensible to me. For trials that only provide a country, only the `facility.address.country` entry would be present. So a complete location entry would look like:

```json
{
  "facility": {
    "name": "{facility name}",
    "address": {
      "street": "{street}",
      "city": "{city}",
      "zip": "{zip}",
      "country": "{country}"
    },
    "location": "{lat}, {lon}",
    "status": "<Not yet recruiting|Recruiting|Active, not recruiting|Completed|Terminated|Suspended|Withdrawn|Enrolling by invitation|Unknown>" ,
    "contact": [
      {
        "use": "{description}",
        "name": "{contact name}",
        "phone": "{phone}",
        "email": "{email}"
      }
    ]
  }
}
```

CT.gov does not have facilities' addresses nor their geolocation, but that data would be very valuable to have. Also, maybe `status` can be a shorter list or just accept any text.

**Comment by vitorbaptista** [comment]
@p2 Sorry for taking long on answering this, I've been travelling for the [https://fda.opentrials.net](OpenTrialsFDA presentation) (vote for us if you haven't already :sweat_smile:) and couldn't give your proposal the time it deserves.

I like it. There're a few details that we have to figure out, and I'm not sure how it'll look like with many registries only offering the country name, but in general I think it's a good approach.

Do you know if the address information is specified like that in the ClinicalTrials.gov XML? Or if we'd have to parse it to extract its parts.

The approach we've been doing for OpenTrials is to gather the data first, and then try to think about a schema. I'm not sure if you've done that already but, if not, a good first step would be to change our [NCT collector](https://github.com/opentrials/collectors/tree/bb6502bd5949612270817e8fc054c68bc4013086/collectors/nct) to extract that information as well, so we can see how the real data looks like and validate the schema against that. What do you think?

I'm going on holiday tomorrow, coming back only by the beginning of January, so won't be responsive. However, the rest of the team will be around and can help you with anything you need.

**Comment by p2** [comment]
Voted!

Same here, a little time constrained myself. Yes, this data is already granularly available from CT.gov; my proposal is closely aligned with what's already available, here's XML straight from ClinicalTrials.gov:

```xml
<location>
  <facility>
    <name>Cantonal Hospital of Grisons, Department of Oncology/ Haematology</name>
    <address>
      <city>Chur</city>
      <zip>7000</zip>
      <country>Switzerland</country>
    </address>
  </facility>
  <status>Recruiting</status>
  <contact>
    <last_name>Roger von Moos, PD</last_name>
    <phone>+41 81 256 6646</phone>
    <email>Roger[...]ch</email>
  </contact>
</location>
```

What I've added is the possibility to add a street name (haven't seen that in CT.gov) and geocoding. I think this could become a field populated by crowdsourcing and cross-referencing.

I haven't done work on the **NCT collector** you describe, but it seems to [already capture CT.gov's `location` property into a `locations` object](https://github.com/opentrials/collectors/blob/master/collectors/nct/parser.py#L236).

---

Another thought is whether facilities could live in their own table/documents and would be linked to from `trial.location`. Since there are many trials at a facility, be that CROs or hospitals, such a separation would remove data duplication, but it would also introduce complexity plus consistency problems (if a facility is renamed or even moves). Probably overkill, but want to capture it here.

**Comment by pwalsh** [comment]
@vitorbaptista I'm on the edge of making this WINTFIX for phase I, but, I really would like to see it in. WDYT?
